lemmy_api/src/comment.rs

```diff
@@ -682,3 +686,156 @@ impl Perform for GetComments {
     Ok(GetCommentsResponse { comments })
   }
 }
+
+#[async_trait::async_trait(?Send)]
+impl Perform for CreateCommentReport {
+  type Response = CreateCommentReportResponse;
+
+  async fn perform(
+    &self,
+    context: &Data<LemmyContext>,
+    websocket_id: Option<ConnectionId>,
+  ) -> Result<CreateCommentReportResponse, LemmyError> {
+    let data: &CreateCommentReport = &self;
+    let user = get_user_from_jwt(&data.auth, context.pool()).await?;
+
+    // check size of report and check for whitespace
+    let reason = data.reason.trim();
+    if reason.is_empty() {
+      return Err(APIError::err("report_reason_required").into());
+    }
+    if reason.len() > 1000 {
+      return Err(APIError::err("report_too_long").into());
```

Comment:
Would be nice if we had some check for untranslated strings. Could be as simple as grepping for `APIError` and checking the value against those in lemmy-translations.

Reply:
Good point... I'll make an issue in lemmy-translations to come up with a script that can do a missing translations check. It would also have to check `lemmy-ui`, because a lot of translations are only for the front end.